docs/defining_systems/si.rst

```diff
@@ -0,0 +1,231 @@
+
+    struct second : named_unit<second, "s"> {};
+    struct dim_time : isq::dim_time<second> {};
+    template<UnitOf<dim_time> U, Representation Rep = double>
+    using time = quantity<dim_time, U, Rep>;
+
+    struct metre : named_unit<metre, "m"> {};
+    struct dim_length : isq::dim_length<metre> {};
+    template<UnitOf<dim_length> U, Representation Rep = double>
+    using length = quantity<dim_length, U, Rep>;
+
+    struct gram : named_unit<gram, "g"> {};
+    struct kilogram : prefixed_unit<kilogram, kilo, gram> {};
+    struct dim_mass : isq::dim_mass<kilogram> {};
+    template<UnitOf<dim_mass> U, Representation Rep = double>
+    using mass = quantity<dim_mass, U, Rep>;
+
+    struct ampere : named_unit<ampere, "A"> {};
+    struct dim_electric_current : isq::dim_electric_current<ampere> {};
+    template<UnitOf<dim_electric_current> U, Representation Rep = double>
+    using electric_current = quantity<dim_electric_current, U, Rep>;
+
+    struct kelvin : named_unit<kelvin, "K"> {};
+    struct dim_thermodynamic_temperature : isq::dim_thermodynamic_temperature<kelvin> {};
+    template<UnitOf<dim_thermodynamic_temperature> U, Representation Rep = double>
+    using thermodynamic_temperature = quantity<dim_thermodynamic_temperature, U, Rep>;
+
+    struct mole : named_unit<mole, "mol"> {};
+    struct dim_amount_of_substance : isq::dim_amount_of_substance<mole> {};
+    template<UnitOf<dim_amount_of_substance> U, Representation Rep = double>
+    using amount_of_substance = quantity<dim_amount_of_substance, U, Rep>;
+
+    struct candela : named_unit<candela, "cd"> {};
+    struct dim_luminous_intensity : isq::dim_luminous_intensity<candela> {};
+    template<UnitOf<dim_luminous_intensity> U, Representation Rep = double>
+    using luminous_intensity = quantity<dim_luminous_intensity, U, Rep>;
+
+    }
+
+
+Derived units
+-------------
+
+The [SIBrochure]_ states:
+
+    Derived units are defined as products of powers of the base units. When the numerical
+    factor of this product is one, the derived units are called coherent derived units. The base
+    and coherent derived units of the SI form a coherent set, designated the set of coherent SI
+    units. The word “coherent” here means that equations between the numerical values of
+    quantities take exactly the same form as the equations between the quantities themselves.
+    Some of the coherent derived units in the SI are given special names. ... Together with the
+    seven base units they form the core of the set of SI units. All other SI units are combinations
+    of some of these 29 units.
+
+
+Unit prefixes
+-------------
+
+According to [SIBrochure]_:
+
+    Prefixes may be used with any of the 29 SI units with special names with
+    the exception of the base unit kilogram.
+
+Here is a complete list of all the :term:`SI` prefixes supported by the library::
+
+    namespace si {
+
+    struct yocto  : prefix<yocto, "y",             mag_power<10, -24>()> {};
```

Comment:
What about these?
```C++
template<PrefixableUnit auto U> struct quecto_ : prefixed_unit<"q", mag_power<10, -30>, U> {};
template<PrefixableUnit auto U> struct ronto_ : prefixed_unit<"r", mag_power<10, -27>, U> {};
```
I think it's worth adding a note on their status.

Reply:
What do you mean? That those are new in SI?

Reply:
Anyway, all the docs will be rewritten from scratch for V2.

Reply:
I don't know. That's why I'd like a note.

Reply:
Here I just merged master to V2 branch with whatever is there. No V2-specific changes applied.

Reply:
Oh, alright. So it's not V2 docs.

Reply:
No, I will start writing docs from scratch soon. Possibly we will also switch to [mkdocs](https://squidfunk.github.io/mkdocs-material/) as I really like the [Au documentation engine](https://aurora-opensource.github.io/au/0.3.2/).

Reply:
V2 is nearly done for gcc-12. Do you think we can put it on mainline as is (after releasing 0.8.0) and provide support for other compilers and docs there on the way, or should those changes be applied to the branch first? I have worked on a branch for nearly a year now, and it sucks ;-):
- many people do not try it before it is not on the mainline
- no Compiler Explorer support
- harder to provide PRs for contributors

Reply:
I'm in favor of [Trunk Based Development](https://trunkbaseddevelopment.com/).
So, yes.

Reply:
It depends on the instantiation context that's the source of the error.
Maybe you can do like `std::view_interface::operator[]`.
See <https://github.com/cplusplus/draft/blob/4fac9f97a2c25d39a01f75cf198d0783bfa8deda/source/ranges.tex#L1738>.

Reply:
This is different. `std::ranges::view_interface::operator[]` does not constrain with the current view type.

Reply:
Yeah. But if the source of the error doesn't actually need the instantiation, it might just work. Otherwise, details of the bug will be necessary to investigate a workaround.